crates/typst/src/symbols/sym.rs

```diff
@@ -495,6 +494,59 @@ pub(crate) const SYM: &[(&str, Symbol)] = symbols! {
         spade.stroked: '♤',
     ],
 
+    // Music.
+    note: [
+        up: '🎜',
+        down: '🎝',
+        whole: '𝅝',
+        half: '𝅗𝅥',
+        quarter: '𝅘𝅥',
+        quarter.alt: '♩',
+        eighth: '𝅘𝅥𝅮',
+        eighth.alt: '♪',
+        eighth.beamed: '♫',
+        sixteenth: '𝅘𝅥𝅯',
+        sixteenth.beamed: '♬',
+        grace: '𝆕',
+        grace.slash: '𝆔',
+        kievan.final: '𝇠',
+        kievan.whole: '𝇢',
+        kievan.half: '𝇣',
+        kievan.quarter.t: '𝇥',
+        kievan.quarter.b: '𝇤',
+        kievan.eighth.t: '𝇧',
+        kievan.eighth.b: '𝇦',
+    ],
+    rest: [
+        whole: '𝄻',
+        multiple: '𝄺',
+        multiple.measure: '𝄩',
+        half: '𝄼',
+        quarter: '𝄽',
+        eighth: '𝄾',
+        sixteenth: '𝄿',
+    ],
+    natural: [
+        '♮',
+        t: '𝄮',
+        b: '𝄯',
+    ],
```

Comment:
`sym.rest` and `sym.natural` aren't obviously tied to music. I think that's a problem.

Reply:
> `sym.rest` and `sym.natural` aren't obviously tied to music. I think that's a problem.

Isn't that an argument that could be made for half the symbols? Does it actually matter unless there's any danger of ambiguity?

Rest and natural are the common names for them.

Reply:
> Isn't that an argument that could be made for half the symbols in sym?

can it? I just scrolled through the file and for most symbols it felt fairly obvious what they are but if I would see `sym.natural` I'd have no clue what it is.

Reply:
AFAIK, LaTeX has `\natural`, `\sharp`, `\flat`.

Reply:
> > Isn't that an argument that could be made for half the symbols in sym?
> 
> can it? I just scrolled through the file and for most symbols it felt fairly obvious what they are but if I would see `sym.natural` I'd have no clue what it is.

This would depend on the reader no?

I guess you could put them under `pitch` or something, as in `pitch.natural`, `pitch.sharp` and `pitch.flat`. Though that makes them quite verbose.

Reply:
Grouping natural/sharp/flat together sounds nice!

Reply:
If we group them, which one shall be the primary symbol? And I don't think `flat.*` (and `sharp.*`) are the variants of `natural.*`.

Reply:
> If we group them, which one shall be the primary symbol? And I don't think `flat.*` (and `sharp.*`) are the variants of `natural.*`.

Does there need to be a primary symbol?

Reply:
In my knowledge to Typst symbols, they always have a primary symbol and may have some variants.

Reply:
> In my knowledge to Typst symbols, they always have a primary symbol and may have some variants.

You already have examples here where that's not the case, like `rest`

Reply:
By "primary symbol", I am wondering what should users get when they use `pitch` (without something like `.natural` following it).

Reply:
> By "primary symbol", I am wondering what users shall get when they use `pitch` (without something like `.natural` following it).

The only logical choice would be natural.

(That being said, I'm not a super-fan of relying on fallback like this)

Reply:
:heart_hands: Done. Please review my newest code.

Reply:
According to Wikipedia, it looks like these things are called [accidentals](https://en.wikipedia.org/wiki/Accidental_(music)).

Reply:
Symbol naming is so hard ... since symbol variants are also not really intended as modules, but more really as variants I'm not 100% sure what to do here. If both of you actually prefer having the three top-level I think I would also be okay with that, given the LaTeX and HTML precedent. (Sorry for the back and forth, I'm just unsure.)

Reply:
> According to Wikipedia, it looks like these things are called [accidentals](https://en.wikipedia.org/wiki/Accidental_(music)).

It's been a long time since I read sheet music, but as far as I understand it's a specific use of them that is known as accidentals. Not if they they're used in a key signature for instance.

Reply:
:heart_hands: Done again. Please review the newest code.